### PR TITLE
addressing #29265

### PR DESCRIPTION
**Pull request by michalsuba**
column name for password string has changed since MySQL 5.7 . This commit adds ability to specify it as a parameter

### CODE REVIEW

File: salt/modules/mysql.py [hunk 1]
```diff
@@ -1080,6 +1080,7 @@ def user_exists(user,
                 host='localhost',
                 password=None,
                 password_hash=None,
+                password_column='Password',
```
Comment (rallytime) [inline salt/modules/mysql.py]: Can you move this line to the end of the `kwarg` statements? We want to avoid an API change here.

**Comment by rallytime** [comment]
@michalsuba Thanks for this change! I made one minor inline comment that should be addressed, but then we can get this in. Thanks!

Refs #29265

**Comment by rallytime** [comment]
Awesome! Thanks!

**Comment by michalsuba** [comment]
Nice, Thank you!

**Pull request merged**